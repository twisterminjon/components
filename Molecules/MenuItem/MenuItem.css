.menuitem {
  position: relative;
  display: flex;
  justify-content: flex-start;
  align-items: center;

  font-size: 16px;
  font-weight: 300 !important;
  font-family: var(--font-stack);
  height: 44px;
  width: 100%;

  color: var(--menu__item_fg);
  background-color: var(--menu__item_bg);

  padding: 0 15px 0 20px;
  text-align: left;
  cursor: pointer;

  /* remove button styles */
  text-transform: none;
  text-shadow: none;
  font-weight: 700;
  line-height: 1em;
  font-style: normal;
  text-decoration: none;
  border: none;
  outline: none;
}

.menuitem:hover {
  background-color: var(--menu__item_hover_bg);
}

.menuitem:focus {
  background-color: var(--menu__item_hover_bg);
}

.menuitem:active {
  background-color: var(--menu__item_active_bg);
}

.menuitem-label {
  flex: 1;
}
