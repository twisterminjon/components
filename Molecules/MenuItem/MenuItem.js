import React from 'react'
import PropTypes from 'prop-types'

import NotificationBadge from '../../Atoms/NotificationBadge/NotificationBadge'

import './MenuItem.css'

MenuItem.propTypes = {
  /** Can show a loading indicator */
  loading: PropTypes.bool,

  /** Label text of item */
  label: PropTypes.string.isRequired,

  /** Can display a badge with a count, if 0 or not provided, badge will be hidden */
  count: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),

  /** Called after the item is clicked */
  onClick: PropTypes.func,

  /** ClassName for the wrapper */
  className: PropTypes.string,

  /** Style for the wrapper */
  style: PropTypes.object,
}

MenuItem.defaultProps = {
  loading: false,
  count: null,
  className: '',
  style: {},
}

export default function MenuItem({ loading, label, count, onClick, className, style }) {
  return (
    <button
      className={`menuitem  ${className} ${loading ? 'button-spinner' : ''}`.trim()}
      onClick={onClick}
      style={style}
      data-testid={`button-menu-${label}`}>
      <span className="menuitem-label">{label}</span>
      {count > 0 && <NotificationBadge count={count} />}
    </button>
  )
}
