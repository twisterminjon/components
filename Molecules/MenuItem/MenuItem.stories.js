import React from 'react'

import { storiesOf } from '@storybook/react'
import centered from '@storybook/addon-centered/react'
import { withKnobs, text, boolean, number } from '@storybook/addon-knobs'
import { action } from '@storybook/addon-actions'

import MenuItem from './MenuItem'

const stories = storiesOf('Provider/v1/Molecules', module)
stories.addDecorator(withKnobs)

stories.addDecorator(centered).add('MenuItem', () => {
  const labelVal = text('label', 'Settings')
  const loadingVal = boolean('loading', false)
  const countVal = number('count', 0)

  return (
    <div style={{ width: '400px' }}>
      <MenuItem label={labelVal} count={countVal} onClick={action('onClick')} loading={loadingVal} />
    </div>
  )
})
