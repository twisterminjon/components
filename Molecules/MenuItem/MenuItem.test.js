import React from 'react'
import MenuItem from './MenuItem'

describe('MenuItem', () => {
  const mockFun = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<MenuItem label="test" onClick={mockFun} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('calls the passed function when clicked', () => {
    const wrapper = window.mount(<MenuItem label="test" onClick={mockFun} />)

    wrapper.find('[data-testid="button-menu-test"]').simulate('click')
    expect(mockFun).toHaveBeenCalled()
  })
})
