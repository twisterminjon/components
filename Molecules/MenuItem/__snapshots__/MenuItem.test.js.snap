// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem matches the snapshot 1`] = `
<button
  className="menuitem"
  data-testid="button-menu-test"
  onClick={[MockFunction]}
  style={Object {}}
>
  <span
    className="menuitem-label"
  >
    test
  </span>
</button>
`;
